koap/src/commonMain/kotlin/Decoder.kt

```diff
@@ -97,6 +99,7 @@ inline fun <reified T : Message> ByteArray.decode(): T =
  * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
  * ```
  */
+@JsExport
```

Comment:
Particular reason for not doing file-wide export?

Reply:
https://github.com/JuulLabs/koap/blob/dtaylor/apd-957_typescript_types/koap/src/commonMain/kotlin/Decoder.kt#L78

> Decoder.kt: (79, 1): Declaration of such kind (inline function with reified type parameters) cant be exported to JS

There are various techniques we can employ to deal with this but I wanted to leave that for a separate PR.